V05release

wildsm opened the pull request:
This address issue #178 

It will follow the release process
https://github.com/bandframework/bandframework/blob/develop/resources/dev_guide/release-proc.rst

mosesyhc committed:
update lcgp sdk link

wildsm committed:
Update README.md

Fixies:
- the link to BFRESCOX
- the version number
- the link to software (so that it tracks the correct branch)
- nuclear physics

wildsm committed:
Update README.md

Fixed some links and version.

wildsm committed:
Update release-proc.rst

Adding three READMEs to the list of places that now contain a version number!

wildsm committed:
Update README.md

Fixed version number and some links.

wildsm committed:
Update README.md

Moved jitr to software consistently

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Merge pull request #156 from bandframework/mosesyhc_lcgp

Initial add of LCGP as submodule

wildsm committed:
Merge branch 'examplereorg' into swpatch1

wildsm committed:
Bringining in latest develop

Bringing in latest develop

DanielRPhillips committed:
Merge pull request #157 from bandframework/examplereorg

Approving pull request of merge of reorganization of Examples directory into develop.

wildsm committed:
Update README.md

It looks like v1.1.0 was still here

wildsm committed:
Update README.md

wildsm committed:
Merge pull request #163 from bandframework/wildsm-patch-1

Update README.md

asemposki committed:
added submodule neutron-rich-bmm to examples folder

beykyle committed:
jitr version update

beykyle committed:
v change

beykyle committed:
update changelog

beykyle committed:
rose version update

kylegodbey committed:
Adding pybmc to the framework

kylegodbey committed:
Added new framework authors

wildsm committed:
Merge branch 'develop' into jitr-version-update

wildsm committed:
Update .wordlist.txt

wildsm committed:
Merge pull request #166 from beykyle/jitr-version-update

Jitr version update

wildsm committed:
Merge branch 'develop' into rose-version-update

wildsm committed:
Update CHANGELOG.rst

beykyle committed:
v1.1.6

wildsm committed:
Update README.md

Fixing URL typo that the CI action caught

wildsm committed:
Update .wordlist.txt

Addressing some new stoplist words

beykyle committed:
one more v update

wildsm committed:
Merge pull request #167 from beykyle/rose-version-update

rose version update

wildsm committed:
Update CHANGELOG.rst

Reflects the changes to develop to date.

wildsm committed:
Update README.md

Updated jitr v here. Also added url for Taweret

DanielRPhillips committed:
Merge pull request #169 from bandframework/wildsm-patch-1

Patching develop based on changes already made

asemposki committed:
Merge pull request #1 from bandframework/develop

Bring up to date with develop

asemposki committed:
adding reade information

wildsm committed:
Update CHANGELOG.rst

wildsm committed:
Update README.md

wildsm committed:
Update README.md

wildsm committed:
Update .wordlist.txt

wildsm committed:
Update README.md

Adding neutron-rich-bmm example

ozgesurer committed:
Update README.md

ozgesurer committed:
Update CHANGELOG.rst

ozgesurer committed:
Update .wordlist.txt

ozgesurer committed:
Update README.md

wildsm committed:
Merge pull request #170 from bandframework/develop-puq

Update README.md

wildsm committed:
Merge pull request #171 from bandframework/develop

Bringing this develop-bmc up to date with develop

wildsm committed:
Adding pybmc SDK to example sdk files

wildsm committed:
Update bandframework-ci.yml

adding pybmc

wildsm committed:
Adding pybmc placeholder to README.md

wildsm committed:
Updating README.md with pybmc current version

wildsm committed:
Adding pybmc to software/README.md

wildsm committed:
Updating CHANGELOG.rst to reflect current pybmc

wildsm committed:
Update AUTHORS

Will address AUTHORS as part of 0.5.0 process

mosesyhc committed:
include surmise update in CHANGELOG.

mosesyhc committed:
spellcheck whitelist update

mosesyhc committed:
brings surmise to v0.4.0

mosesyhc committed:
update README to upgrade surmise to v0.4

mosesyhc committed:
SDK update

mosesyhc committed:
use Jupyter Book as separate words

mosesyhc committed:
point to most updated surmise with upgrade instructions

wildsm committed:
Merge pull request #172 from bandframework/surmise-v0.4-update

include surmise v0.4 in update

kylegodbey committed:
Update pybmc version in CHANGELOG

kylegodbey committed:
Update pybmc version in README.md

kylegodbey committed:
Update pybmc version to v0.2.4 in README

kylegodbey committed:
Update pybmc version in README to v0.2.4

wildsm committed:
Merge pull request #168 from bandframework/develop-pybmc

Add pybmc as Software

asemposki committed:
updating submodule SAMBA

wildsm committed:
Merge branch 'develop' into develop

asemposki committed:
updating neutron-rich-bmm with data and SDK renaming

kylegodbey committed:
Update BMEX description with version info

Added version information for the BAND release to the BMEX description.

kylegodbey committed:
Fix documentation path and add version info

Corrected the file path for BAND SDK compatibility documentation and added version information.

kylegodbey committed:
Update citation details in README.md

kylegodbey committed:
Fix capitalization in BMEX description version label

asemposki committed:
updating submodule and new release of neutron-rich-bmm

asemposki committed:
Update neutron-rich-bmm version in CHANGELOG

Updated the version of neutron-rich-bmm from v0.0.1 to v0.1.0 in the changelog.

asemposki committed:
updating version of submodule neutron-rich-bmm

asemposki committed:
updating the neutron-rich-bmm link in README

kylegodbey committed:
Merge remote-tracking branch 'origin/develop' into develop-bmex

kylegodbey committed:
Update Changelog to include BMEX

asemposki committed:
updating submodule SAMBA for readme changes

asemposki committed:
trying to merge so we can update

kylegodbey committed:
Fix formatting of description in BMEX-bandsdk.md

kylegodbey committed:
Fix formatting in BMEX README.md

DanielRPhillips committed:
Merge pull request #173 from bandframework/develop-bmex

Update BMEX BAND Example

wildsm committed:
Merge pull request #165 from asemposki/develop

Adding submodule neutron-rich-bmm to examples folder

sjaiswal-tifr committed:
added submodule

sjaiswal-tifr committed:
added submodule ModelDiscrepancy

wildsm committed:
Update README with ModelDiscrepancy example

Added ModelDiscrepancy example to README.

wildsm committed:
Update README with ModelDiscrepancy example

Added a new example entry for ModelDiscrepancy with a link to its repository.

wildsm committed:
Update CHANGELOG with MD

wildsm committed:
Add link to MD-bandsdk.md in README

wildsm committed:
Add 'ModelDiscrepancy' to wordlist

wildsm committed:
Update bandframework-ci.yml

wildsm committed:
Updated examples/README

CI, after it was updated to reflect new BAND structure) caught a missing space.

asemposki committed:
updating submodule Taweret for v0.5

asemposki committed:
Merge branch 'bandframework:develop' into develop

wildsm committed:
Update AUTHORS with new contributors

Added new contributors to the AUTHORS file based on all confirming.

wildsm committed:
Update author list in citation for BAND Framework

asemposki committed:
updating submodule with new data

asemposki committed:
Merge branch 'develop' of https://github.com/asemposki/bandframework into develop

wildsm committed:
Merge pull request #175 from sjaiswal-tifr/develop

Adding submodule ModelDiscrepancy to examples folder

wildsm committed:
Merge pull request #177 from bandframework/v0.5.0-Author-update

Update AUTHORS with new contributors

asemposki committed:
Merge pull request #2 from bandframework/develop

Bringing up to date with Develop

wildsm committed:
Taweret in CHANGELOG

Updated various BAND-compatible packages to their latest versions, including bug fixes and new features.

wildsm committed:
Update Taweret version in README.md

wildsm committed:
Update Taweret version to v1.2.0 in README

asemposki committed:
updating Taweret submodule

asemposki committed:
adding some ninja clarification

wildsm committed:
Merge pull request #176 from asemposki/develop

Updating submodule Taweret to v1.2.0

wildsm committed:
Update README.md links to reflect new paths

Addressing #174

wildsm committed:
Fix links to use consistent BFRESCOX casing

Addressing #174

wildsm committed:
Update README link for frescox documentation

Addressing #174

wildsm committed:
Merge pull request #179 from bandframework/bfrescoxv05

Bfrescox link updates for bandframework v05

wildsm committed:
Merge pull request #180 from bandframework/develop

Bringing up to date with develop

wildsm committed:
Update README for version 0.5.0 changes

wildsm committed:
Update authors list for BAND Framework to v0.5.0

wildsm committed:
Revise citation details and version number

Updated author list and version number for BAND Framework citation.

wildsm committed:
Update README to reflect version change to 0.5.0

wildsm committed:
Update README to reflect package version change

wildsm committed:
Update README for BAND software version change

beykyle committed:
add bfrescox

beykyle committed:
add bfrescox to readme

beykyle committed:
Merge branch 'v05release' into bfrescox

beykyle committed:
capitalization

beykyle committed:
merge

beykyle committed:
stefan & jared's review

DanielRPhillips committed:
Jared's suggestion for update of Bfrescox BANDsoftware_uses link

ozgesurer committed:
Create bandcamp2021_nbs

ozgesurer committed:
including nbs

ozgesurer committed:
Update BANDCamp21.rst

ozgesurer committed:
Update BANDCamp21.rst

ozgesurer committed:
Update BANDCamp21.rst

ozgesurer committed:
Update BANDCamp21.rst

ozgesurer committed:
Update README.rst

ozgesurer committed:
Update README.rst

ozgesurer committed:
Update README.rst

ozgesurer committed:
Update BANDCamp21.rst

wildsm committed:
Update authors list for BAND Framework v0.4.0

wildsm committed:
Fix typo in AUTHORS file version string

wildsm committed:
Update README.md

wildsm committed:
Update README for versioning changes to 0.4.0dev

wildsm committed:
Update version number in README.md

wildsm committed:
Update version number in bandsdk.md

wildsm committed:
Update version information in README.md

wildsm committed:
Update publication year to 2024

wildsm committed:
Refactor author list in BAND Framework citation

Condensed the author list in the BAND Framework citation.

wildsm committed:
Fix formatting of Bfrescox URL - caught by actions

wildsm committed:
Correct capitalization of 'Bfrescox' in README

wildsm committed:
Add 'Bfrescoxbandsdk' to wordlist

wildsm committed:
Fix formatting of Bfrescox link caught by CI/Actions

beykyle committed:
capitalize Bfrescox

beykyle committed:
Stefan's comments

wildsm committed:
Fix formatting for Bfrescox changelog

DanielRPhillips committed:
Merge pull request #181 from beykyle/bfrescox

Two reviews have been completed, merging this into develop.

DanielRPhillips committed:
Revert "Bfrescox"

wildsm committed:
Merge pull request #183 from bandframework/revert-181-bfrescox

Revert "Bfrescox"

beykyle committed:
add bfrescox at correct version

beykyle committed:
put snippets back into snippet places

beykyle committed:
fix URLs

beykyle committed:
adding space before end of URL, thank you Stefan

beykyle committed:
fix typos

beykyle committed:
fix URL?

beykyle committed:
whitelist spelling

wildsm committed:
Merge pull request #185 from beykyle/bfrescox

Bfrescox

wildsm committed:
Merge pull request #186 from bandframework/develop

Bringing up to date with new develop

wildsm committed:
Update CHANGELOG 

Added new BAND-compatible tools and updated existing ones, including Bfrescox and various other packages.

ozgesurer committed:
Merge pull request #182 from bandframework/bfresco-tutorial

Bfresco tutorial

wildsm committed:
Merge pull request #187 from bandframework/develop

Bringing release up to date with develop

wildsm committed:
Update CHANGELOG for release 0.5.0

Updated the release date for version 0.5.0 and consistent capitalization

jared321 commented:
Did we lose some changes to the root README with the Bfrescox revert?
https://github.com/bandframework/bandframework/pull/183/files

It looks like the current README doesn't include updates to the Bfrescox use example

> [Bfrescox + surmise](/BANDsoftware_uses/Bfrescox): An example of using the Frescox scattering code with surmise to enable Bayesian parameter estimation for coupled-channels scattering.

and Bfrescox's submodule is no longer included at the end of the list of framework software

> [software/](software/)Bfrescox

wildsm commented:
@jared321 - excellent spot. Can you please update those on this branch?

jared321 committed:
Revert reverted Bfrescox content

Updated README to include Bfrescox with surmise example.

jared321 commented:
@wildsm I see the following placeholder versions
```
./BANDsoftware_uses/README.md:This directory contains examples of the application of the BAND tools, found [here](../software), to nuclear physics problems. As of v0.4.0+dev we have the following examples:
./resources/sdkpolicies/bandsdk.md:        number      = {Version 0.4.0+dev},
./software/README.md:As of v0.4.0+dev the following packages are present in this directory. 
./README.md:As of version 0.4.0+dev, the following tools are included:
./README.md:The following [examples](/examples/) are included in version 0.4.0+dev:
./README.md:        number      = {Version 0.4.0+dev},
./examples/README.md:As of 0.4.0+dev this directory includes the following examples: 
```
Do we want those changed to `0.5.0` before the review?

wildsm committed:
Update version number to 0.5.0 in README

wildsm committed:
Update authors list for BAND Framework v0.5.0

wildsm committed:
Revise citation details in bandsdk.md

Updated author list, version number, and year for the BAND Framework citation.

wildsm committed:
Update README to reflect version 0.5.0 changes

wildsm committed:
Update README to reflect version change to v0.5.0

wildsm committed:
Update README to reflect version change to v0.5.0

wildsm committed:
Update CHANGELOG with some known issues

Added known issues regarding AI tools and BAND package installation.

DanielRPhillips committed:
Added Known issue regarding Bfrescox in BANDsoftware_uses

DanielRPhillips committed:
Added another author

DanielRPhillips committed:
Update AUTHORS

DanielRPhillips committed:
Added additional author to author list in SDK

wildsm commented:
Reviewers, this is now ready for you to look at. At a minimum, please do the following:

- [ ] Release notes for this version are complete and represent changes through the `CHANGELOG.rst </CHANGELOG.rst>`_ file.

- [ ] Ensure that links and references have been updated (e.g., no occurrences of ``privateband``). Consider checking that the url checking action is reviewing a sufficiently rich set of files where spelling should be checked.

- [ ] Version number is update-to-date wherever it appears and ``+dev`` suffix is removed
  **(currently: 3 in `README.md </README.md>`_, 1 in `AUTHORS </AUTHORS>`_, 1 in `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_, 1 in `examples </examples/README.md>`_, 1 in `software </software/README.md>`_, 1 in `BANDsoftware_uses </BANDsoftware_uses/README.md>`_, and possibly in `CHANGELOG.rst </CHANGELOG.rst>`_)**.
  
- [ ] Check `README.md </README.md>`_ *Citing bandframework* for correctness (e.g., ensure that author list matches `AUTHORS </AUTHORS>`_, **year and version are correct**).

- [ ] Check `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_ *Citing bandframework* for correctness **(authors, year, version)**. Ensure that SDK version listed is the same as in the provided `template </resources/sdkpolicies/template.md>`_.

- [ ] Check `sdkpolicies README </resources/sdkpolicies/README.md>`_ for correctness (e.g., ensure that each software in the release has a working link for its SDK compliance, also note any differences in the version of the SDK that each package lists versus the version at `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_).

- [ ] Please state whether you have verified successful execution of the notebooks linked in `BRICK </examples/BRICK/tutorial>`_ and `Bfrescox use </BANDsoftware_uses/BFRESCOX>`_ . Successful completion of this PR should include verification across the union of reviewers.

- [ ] Tests are run with source to be released (this may iterate):

  - Online CI (GitHub Actions) tests must pass.

  - Documentation must build and display correctly wherever hosted.


**When finished, please approve, but please do not merge.**

jared321 commented on BANDsoftware_uses/README.md:
```diff
@@ -0,0 +1,4 @@
+This directory contains examples of the application of the BAND tools, found [here](../software), to nuclear physics problems. As of v0.5.0 we have the following examples:
+
+- [Bfrescox](BFRESCOX/README.md): A BAND extension of the frescox scattering code for coupled-channels calculations that uses surmise to enable parameter estimation within frescox.
```
Should this text, including the link name, be identical to the analogous description in the root README?

Does it need to be here?  See comments for README.md in "examples" folder.

wildsm replied:
IMO, we do not need complete copies, but the distinctions should be deliberate. @ozgesurer @DanielRPhillips do you prefer the text here or in the root README? Want to align?

ozgesurer replied:
Update the folder name: BFRESCOX --> Bfrescox

DanielRPhillips replied:
I somewhat aligned the two, but--following @wildsm's suggestion--made the description of the directory a little bit different in the two places. The way the BANDsoftware use example is described is the same in both places. We should probably make sure the two descriptions are aligned for other BANDsoftware use cases added to this directory in the future.

jared321 commented on CHANGELOG.rst:
```diff
@@ -6,6 +6,37 @@ Below are the release notes for all bandframework releases.
+
+New capabilities and notable changes:
+
+- added BAND-compatible lcgp at `v0.2.1 <https://github.com/mosesyhc/LCGP/releases/tag/v0.2.1>`_, a tool for latent component Gaussian process emulation for multivariate stochastic simulations.
```
The repo refers to the package as "LCGP" rather than "lcgp".

wildsm replied:
@mosesyhc - do you have a preference? The docs and repo use LCGP, while the package is installed as lcgp. 

We should at least be internally consistent within the bandframework repo.

jared321 replied:
@mosesyhc Confirmed that he prefers LCGP to lcgp.  Change made at b02aa51.

jared321 commented on CHANGELOG.rst:
```diff
@@ -6,6 +6,37 @@ Below are the release notes for all bandframework releases.
+New capabilities and notable changes:
+
+- added BAND-compatible lcgp at `v0.2.1 <https://github.com/mosesyhc/LCGP/releases/tag/v0.2.1>`_, a tool for latent component Gaussian process emulation for multivariate stochastic simulations.
+- added BAND-compatible ModelDiscrepancy at `v1.1.0 <https://github.com/sjaiswal-tifr/ModelDiscrepancy/tree/32d7d2c46009bc8c67a86580a088fb1b747ae029>`_, an example Bayesian framework for model-data comparison that accounts for theoretical uncertainties.
```
This link refers to a particular commit by hash rather than by tag.  All other links point toward the release notes.

wildsm replied:
Addressed in https://github.com/bandframework/bandframework/commit/ea3d951fa89c1a6ce811004af8f10172dace8a80

jared321 commented on examples/README.md:
```diff
@@ -1,4 +1,13 @@
+
+As of 0.5.0 this directory includes the following examples: 
+
+- [BMEX](BMEX/): A web application for exploring quantified theoretical model predictions of nuclear masses and related quantities.
```
Why do we need to list the contents of this folder in this README?  If visitors would like to see what is included and get more information about each, they can just click on each folder/link.

If the point is to capture the state of those folders/links at v0.5.0, then hasn't this already been done in the root README?  Ensuring correct contents and links in so many places seems difficult.

wildsm replied:
The design principle here reflects a desire to have top-level ("element"+ !) directories having something explain what is there. 

Indeed this could point back to the root README so as to have a single place to maintain, but it is helpful to have subTOCs when walking through the elements of the band framework. 

@jared321 - Do you think adding to each of these one-layer-below-root READMEs a link to the README would help?

jared321 replied:
There is a lot of redundancy in the documentation.  For instance, you have information about (including links to) the contents of the current release in
* the bandframework repo's release notes
* the root README
* the READMEs in the different subfolders

The first two make sense to me and I can appreciate how the third can be helpful to users.  But, under the assumption that users are reading the subTOCs through the GH web interface with the same list above the README rendering, that duplication seems unnecessary.  Could we get rid of the links in the subTOCs for example?

Related to this it might be good to point out to users in the subfolder READMEs what the submodule hashes correspond to.  Are they the version of that element in the latest BAND framework release?  Can they be updated between releases on `main`?  If not, it might be good in the root and subfolder READMEs to point out that the 
 versions of the individual elements can be updated between BAND framework releases.

wildsm replied:
Removing links in the main three subfolder READMEs:

https://github.com/bandframework/bandframework/commit/1ff1b71b6b32b1999d4cefe09b899c3c0a12f70f 

https://github.com/bandframework/bandframework/commit/80946a1be21e2410f85c99f4bf83c2f1ff0cec5d

https://github.com/bandframework/bandframework/commit/d806c574fbd63570eee0110ea3bb31131b103011

https://github.com/bandframework/bandframework/commit/292077af998394eadc419af7b084d836599e0a21

Regarding second point, I think it's good, but am looking for a suggestion.

DanielRPhillips replied:
If I understand this conversation correctly two useful changes here would be:
- Have each subfolder README link back to the main README file
- State explicitly that if software is included by submodule then we do not commit to updating the hash between framework releases, so a more recent version of the software may be available

I can make those two changes. 

I do agree with @wildsm that the subTOCs are useful to someone who is navigating through, so I am disinclined to _only_ have the link to the main README. But I also agree with @jared321 that the links for the subTOCs are right there above the README, so we could make our lives simpler w/o sacrificing very much guidance for users if we removed the links in the subTOC. Thanks for making that change @wildsm

DanielRPhillips replied:
I have made the two changes in the bulleted list of my previous comment.

jared321 commented on examples/README.md:
```diff
@@ -1,4 +1,13 @@
+- [BMEX](BMEX/): A web application for exploring quantified theoretical model predictions of nuclear masses and related quantities.
+- [BRICK](BRICK/): The Bayesian R-matrix Inference Code Kit, designed to facilitate extraction of R-matrix parameters from experimental data.
+- ModelDiscrepancy ([v1.1.0](https://github.com/sjaiswal-tifr/ModelDiscrepancy/tree/32d7d2c46009bc8c67a86580a088fb1b747ae029 )): A Bayesian framework for model-data comparison that accounts for theoretical uncertainties.
+- [neutron-rich-bmm](https://github.com/asemposki/neutron-rich-bmm/tree/07054335346e4a26ed919967e818cccd2d52c7ef): A Gaussian process Bayesian model mixing approach for microscopic constraints for the equation of state and structure of neutron stars.
```
neutron-rich-bmm is versioned.  Indeed, the ink corresponds to v0.1.0.

wildsm replied:
Confirmed and addressed in https://github.com/bandframework/bandframework/commit/7edef8292e2b27e0486c1422c5f51a4d5f554943

jared321 commented on examples/README.md:
```diff
@@ -1,4 +1,13 @@
+- [neutron-rich-bmm](https://github.com/asemposki/neutron-rich-bmm/tree/07054335346e4a26ed919967e818cccd2d52c7ef): A Gaussian process Bayesian model mixing approach for microscopic constraints for the equation of state and structure of neutron stars.
+- [nsat](https://github.com/cdrischler/nuclear_saturation/tree/06cf466c2ab5f6e1fccafc18807c7dd99aff05c7): A Bayesian mixture model approach to quantifying the empirical nuclear saturation point.
+- [QGP_Bayes](https://github.com/danOSU/QGP_Bayes/tree/4b3e2364f87a29ad2469f2b072053420fdaac8e9): A tutorial on the use of JETSCAPE_SIMS tools to infer parameters of the QGP.
+- SaMBA ([v1.2.0](https://github.com/asemposki/SAMBA/tree/3f255109624be5fa5e761f4acfad35b9d61a00d1 )): The Sandbox for Mixing via Bayesian Analysis.
```
The link points to the latest commit on `main`.  However, this is different from the commit associated with v1.2.0 (fc181b23).  Is this intentional?

wildsm replied:
@asemposki - I believe this may have been done because of a very minor change post v1.2.0. If so, what do you think about dropping the version number here and linking SaMBA with the same url but w/o a version number (similar to QGP_Bayes above)? The next question would be about consistency for root README and CHANGELOG.

jared321 commented on examples/README.md:
```diff
@@ -1,4 +1,13 @@
 # bandframework examples
```
BAND framework examples?

wildsm replied:
It does look like 
```
grep -r "bandframework " .
```

and

```
grep -ir "band f" .
```

reveal inconsistent branding.

Do others have opinions on the importance?

We could at least move to reduce the number of the former occurrences that directly appear.
I will start to do this, but it would be good to hear from others,

jared321 replied:
My two cents:
* I would assume that "bandframework" refers to this repository.  For instance, you could talk about "bandframework policies" related to working in this repository.
* "BAND Framework" is what is made public via the repos in this organization and by repos outside the organization.  We seem to use the latter fairly consistently in the root README and the web page.

DanielRPhillips replied:
@jared321: what you said is the way I think about it too. The BAND Framework is the thing we are building. bandframework is where it is built.

DanielRPhillips replied:
All four README's (main, examples, BANDsoftware_uses, software) are now updated to this usage convention.

jared321 commented on resources/sdkpolicies/README.md:
```diff
@@ -11,11 +11,15 @@ A parallel [directory](/resources/dev_guide) contains a basic development guide
 -  [BMEX-bandsdk.md](/software/BMEX/BMEX-bandsdk.md)
 -  [brickbandsdk.md](/software/BRICK/brickbandsdk.md)
--  [frescoxbandsdk.md](/software/Bfrescox/frescoxbandsdk.md)
+-  [bfrescoxbandsdk.md](https://github.com/bandframework/Bfrescox/blob/v0.0.1-alpha/Bfrescoxbandsdk.md)
```
Note that this link is presently hardcoded to a release, which makes sense since the correct version of the SDK doc is presently on a feature branch.  In the future, this should be edited to refer to the file on `main` as done for others.

Ask Bfrescox to include the updating of this link as part of post-release tasks?

wildsm replied:
@beykyle please review that you are OK with 
https://github.com/bandframework/bandframework/commit/7cbd2cdf163a0efd6c6a0fc141f2ed5d2196c80d 
and the associated issue generated at
https://github.com/bandframework/Bfrescox/issues/42

beykyle replied:
This all makes sense to me.

jared321 commented on resources/sdkpolicies/README.md:
```diff
@@ -11,11 +11,15 @@ A parallel [directory](/resources/dev_guide) contains a basic development guide
 -  [jitrbandsdk.md](https://github.com/beykyle/jitr/blob/main/jitrbandsdk.md)
+-  [lcgp-bandsdk.md](https://github.com/mosesyhc/LCGP/blob/main/lcgp-bandsdk.md)
+-  [MD-bandsdk.md](https://github.com/sjaiswal-tifr/ModelDiscrepancy/blob/main/MD-bandsdk.md)
+-  [neutron-rich-bmm_bandsdk.md](https://github.com/asemposki/neutron-rich-bmm/blob/main/neutron-rich-bmm_bandsdk.md)
```
We are including SDK for "examples", which are not officially part of the framework.  Do they really need SDK docs?  We are no longer, however, linking to the Bfrescox+surmise BAND use example SDK.

I thought that only BAND software and BAND use examples might need this.  If this is the case maybe we should make it clear in the resources section that additions to "examples" don't require this extra effort.

wildsm replied:
Readded frescoxbandsdk (from the software uses Bfrescox) via https://github.com/bandframework/bandframework/commit/59335467da4db47d03ed5e7a4554e9e59943f80a 

I believe it is good to list as broad of a range  of example sdk docs at this location, especially as they continue to be reviewed. 

Good point to note the distinction, I am surprised it did not appear elsewhere. I have made a proposal via https://github.com/bandframework/bandframework/commit/9673bdf4e7cf58e03504c8e479fdd38ef318ea33

jared321 replied:
I like the new text.  I suspect that the "frescoxbandsdk" link was broken by a later rename of the folder.

DanielRPhillips replied:
I also like the new text. Rename of folder was indeed culprit in dead frescoxbandsdk link. I have fixed that now.

jared321 commented on resources/sdkpolicies/bandsdk.md:
```diff
@@ -54,13 +54,14 @@ Please use the following to cite the BAND Framework:
 
     @techreport{bandframework,
```
Why is this buried deep in this folder hierarchy/file?  The requested citation is already given in the root README.

wildsm replied:
The intent here is a re-amplification of the citation for those who come to BAND via the SDK. 

IMO it should appear (i.e., not just have a link back to root README). Other thoughts?

@jared321 - if we wanted to maintain a single bibtex source, is there a good way that you know of for these to simply appear in both places while edited in one?

jared321 replied:
A quick search suggests that you cannot insert the contents on one file into the rendering of a markdown file.  Rather, this is something that could be done by scripts/actions.

I have not done this before, but it would make sense for the repo to have something like a JSON file that contains the definition of each release.  Perhaps scripts could then be used to insert correct release-related content into the different READMEs.  This could include an author list, for example.

jared321 commented on software/README.md:
```diff
@@ -1,4 +1,19 @@
+**Read [bandframework/resources/bandsdk](/resources/sdkpolicies/) and [CONTRIBUTING](CONTRIBUTING.rst) before attempting to contribute to
+this directory.**
+
+As of v0.5.0 the following packages are present in this directory. 
```
LCGP is not listed.

As for the "examples" README, I believe that this list is redundant and increases maintenance difficulties.

wildsm replied:
See above comments on multiple READMEs, but definitely needed to add LCGP and this also pointed out that we should alphabetize consistent with root README.

Changes addressed in https://github.com/bandframework/bandframework/commit/ca5ecab2a38d6565babb2d5aa538108acfd7a7e0

jared321 commented on software/README.md:
```diff
@@ -1,4 +1,19 @@
-this branch.
+This contains the core software tools for the bandframework. 
+
+**Read [bandframework/resources/bandsdk](/resources/sdkpolicies/) and [CONTRIBUTING](CONTRIBUTING.rst) before attempting to contribute to
```
The CONTRIBUTING link is broken.  If we have a global CONTRIBUTING doc, shouldn't it direct potential contributors to the SDK policy section?

wildsm replied:
Fixed broken link via https://github.com/bandframework/bandframework/commit/b998c827cf3376d77783d51886ee64288af6a423

Surprised this was not caught via https://github.com/bandframework/bandframework/blob/v05release/.github/workflows/bandframework-ci.yml , which does include software/README.md

https://github.com/bandframework/bandframework/blob/v05release/CONTRIBUTING.rst does direct people to https://github.com/bandframework/bandframework/blob/v05release/resources/sdkpolicies/bandsdk.md , please note if you had something else in mind.

jared321 replied:
@wildsm  In that case, pointing potential contributors to just the larger scope CONTRIBUTING doc should be sufficient and clearer.  Feel free to mark as resolved after reading.

wildsm replied:
I didn't want to pass up the emphasis on the sdk, but have changed the order to clarify that CONTRIBUTING is the authority. https://github.com/bandframework/bandframework/commit/199ad83a44443f870b7f8d6063489d9ce93b633a

jared321 commented on software/rose:
`rose` is listed as included in this release.  Why was its link deleted?

wildsm replied:
Excellent catch. @DanielRPhillips ?

DanielRPhillips replied:
Yup. That's a problem. Somewhere along the various commits to develop the rose submodule got dropped. I have added it back in via https://github.com/bandframework/bandframework/commit/878ac22c238970f3908164275d7d69d62d55ef73

jared321 commented on README.md:
```diff
@@ -29,30 +29,36 @@ A full list of BAND members together with a current list of  publications produc
 
-- [Bfrescox](/software/Bfrescox): A BAND extension of the frescox scattering code for coupled-channels calculations that uses surmise.
+The following [examples](/examples/) are included in version 0.5.0:
 - BMEX ([v0.1.1](https://github.com/massexplorer/bmex-masses/releases/tag/v0.1.1 )): A web application for exploring quantified theoretical model predictions of nuclear masses and related quantities.
```
The similar line in the "examples" README does not indicate that BMEX is versioned.  v0.1.1 is from 2022, whereas [v0.1.4 was created recently](https://github.com/massexplorer/bmex-masses/releases/tag/v0.1.4).

wildsm replied:
Updated to v0.1.4, which was part of review, CHANGELOG, etc.

Also alphabetized software.

Changes in https://github.com/bandframework/bandframework/commit/d11ddcf194fddfca462d169f792e7e331a25baa5

jared321 replied:
Should the BMEX entry in the "examples" README be updated to show the version as well?

DanielRPhillips replied:
It looks like the "examples" README points to the BMEX and BRICK subdirectories within examples, i.e., to something that resides in our bandframework/bandframework repo. To my way of thinking this is a different case to examples that are submoduled in.  In those cases we should specify which version of the software we are trying to point to.

wildsm replied:
Note that the examples README no longer has links to examples

DanielRPhillips replied:
In order to ensure consistency across different examples I added version number (v0.1.4) for BMEX. Verified that all examples for which version numbers are not listed are unversioned.

jared321 commented on README.md:
```diff
@@ -29,30 +29,36 @@ A full list of BAND members together with a current list of  publications produc
 - [nsat](https://github.com/cdrischler/nuclear_saturation/tree/c4cfa45a1180b2739e217102d7380736d6844a11): A Bayesian mixture model approach to quantifying the empirical nuclear saturation point.
 - [QGP_Bayes](https://github.com/danOSU/QGP_Bayes/tree/4b3e2364f87a29ad2469f2b072053420fdaac8e9): A tutorial on the use of JETSCAPE_SIMS tools to infer parameters of the QGP.
-- SaMBA ([v1.1.0](https://github.com/asemposki/SAMBA/releases/tag/v1.1.0 )): The Sandbox for Mixing via Bayesian Analysis.
+- SaMBA ([v1.2.0](https://github.com/asemposki/SAMBA/tree/3f255109624be5fa5e761f4acfad35b9d61a00d1 )): The Sandbox for Mixing via Bayesian Analysis.
```
ModelDiscrepancy and SAMBA point to a particular commit (not necessarily a release commit) rather than the release notes.

wildsm replied:
For SAMBA, the version in v0.5.0 is actually slightly past the tagged release v1.2.0: e.g., it made updates to respond to BAND reviews: https://github.com/asemposki/SAMBA/commit/9fa6c55d7966457e271b3e2a660eba4bcc1dcd6b

@jared321 & @asemposki - One possible fix would be to link here (and in examples README, and maybe CHANGELOG?) the current url without a version number (e.g., as done for QGP_Bayes). 

ModelDiscrepancy has been fixed via https://github.com/bandframework/bandframework/commit/afa46fb61b152bae51d615de8a7ed40c1b99f1d6 and https://github.com/bandframework/bandframework/commit/f04e20faa36882f0bfbc2029f9dc0a5e344e3c36

asemposki replied:
I think that the link should be changed to the most recent commit. I made a lot of documentation formatting related changes that aren't a worthwhile new version but also are important for rendering the docs correctly in v0.5, so it should be that commit that is linked.

asemposki replied:
I can change the URL in each of the places @wildsm mentions above to reflect the latest commit.

DanielRPhillips replied:
Thanks. Please go ahead and do that. While you're at, maybe also remove the SAMBA version number where it occurs?  Unless you think it's important that the version number be listed.

asemposki replied:
OK instead of removing the version, I have just made a new version v1.2.1 which incorporates the documentation-level changes as a small release. This makes sense given my structuring of releases, so I went ahead and did that. Now I will update the tags to v1.2.1 and every URL to the latest commit on main or release notes where applicable.

asemposki replied:
I have made the requested changes on this branch below.

jared321 commented on README.md:
```diff
@@ -63,13 +69,15 @@ You are free to use any pieces of the BAND Framework that will advance your own
 
-* [software/](software/)nuclear_saturation
-* [software/](software/)QGP_Bayes
+* [examples/](examples/)nuclear_saturation
```
Instead of maintaining this list, which is likely not so important for this section, can you just point readers to the two folders that contain submodules?

wildsm replied:
This is an excellent suggestion.  Addressed in https://github.com/bandframework/bandframework/commit/d0ddc28d3d3018a7bab2368293062206bcd0b6e2

jared321 commented on README.md:
```diff
@@ -29,30 +29,36 @@ A full list of BAND members together with a current list of  publications produc
 
 ## BAND Framework Elements
```
How about "BAND Framework Elements v0.5.0" so that you don't have to keep mentioning v0.5.0 below.  For example, to be consistent with what is currently written, you should also add "As of version 0.5.0" to the BAND software uses line below.

wildsm replied:
The latter has been done for consistency in https://github.com/bandframework/bandframework/commit/efd78d3dd7f0ee45e04f404a45dea5da8682f631

Decided not to add a version to the header since the open paragraph aspires to be more general.

jared321 replied:
It looks like this was resolved by adding "As of version...".  Please mark as resolved if this is the case.

DanielRPhillips replied:
I agree. That takes care of it. Thanks both of you.

jared321 commented on README.md:
```diff
@@ -29,30 +29,36 @@ A full list of BAND members together with a current list of  publications produc
+- [SmoothEmulator](/software/SmoothEmulator): A simplex sampler, emulator trainer, and MCMC explorer that employs a smooth emulator.
+- PUQ ([v0.1.1](https://github.com/parallelUQ/PUQ/releases/tag/v0.1.1 )): A Python package for generating experimental designs tailored for uncertainty quantification and featuring parallel implementations.
+- jitr ([v2.5.1](https://github.com/beykyle/jitr/releases/tag/v2.5.1 )): A Python package containing a Lagrange mesh R-matrix solver for parametric reaction model calibration.
+- lcgp ([v0.2.1](https://github.com/mosesyhc/lcgp/releases/tag/v0.2.1 )): A Gaussian process surrogate model for emulating stochastic simulation outputs.
```
This package is referred to as "LCGP" as opposed to "lcgp" throughout its repo.

wildsm replied:
@mosesyhc - This is a second comment regarding the one I tagged you in above

jared321 replied:
@mosesyhc Confirmed that he prefers LCGP to lcgp.  Change made at b32cd0a.

jared321 reviewed (commented):
I have reviewed all the changes made.  Please see comments and questions.  I'll resume the review after things are (potentially) cleaned.

Other questions:

- Has anyone run the Bfrescox and BRICK notebooks after the move?
- The license section in the root README specifically mentions the "software" folder.  Should it also mention the "examples" and "BANDsoftware_uses" folders as well?

wildsm committed:
Simplify submodule usage in README

Updated the README to clarify the use of submodules.

wildsm committed:
Fix link for ModelDiscrepancy in CHANGELOG

Making consistent with other links to tagged releases

wildsm committed:
Update README to include examples and uses in license section

wildsm commented:
> I have reviewed all the changes made. Please see comments and questions. I'll resume the review after things are (potentially) cleaned.
> 
> Other questions:
> 
>     * Has anyone run the Bfrescox and BRICK notebooks after the move?
> 
>     * The license section in the root README specifically mentions the "software" folder.  Should it also mention the "examples" and "BANDsoftware_uses" folders as well?

Regarding notebooks, I think this is an important point and suggest that the reviewers ensure that they have collectively covered these.  I am adding this to the review template for the PR. I am presuming you are referencing the Bfrescox BANDsoftware_uses notebooks, but please flag this if I misunderstood.

The license section has been updated via https://github.com/bandframework/bandframework/commit/7eb895def55cc7061210da4d5be47bcc1ce0b38b .

wildsm committed:
Update link for bfrescoxbandsdk.md to main branch

For consistency. Added issue.

wildsm committed:
Update neutron-rich-bmm entry in README

wildsm committed:
Update README with LCGP and alphabetizing

Responding to #188

wildsm committed:
Fix link formatting in README.md

Addressing #188

wildsm committed:
Update BMEX version in README & alphabetize software

Updated BMEX version from 0.1.1 to 0.1.4 to reflect what was reviewed. Addressing #188

wildsm committed:
Readded link to frescoxbandsdk.md in SDK examples

wildsm committed:
Update README with BAND SDK document requirements

Refined clarity around requirements for BAND SDK Compatibility Document.

ozgesurer committed:
folder tmp name

ozgesurer committed:
lower case Bfrescox

ozgesurer committed:
fix folder name in other files

ozgesurer committed:
Merge pull request #190 from bandframework/v05_bfresco_fix

V05 bfresco fix

wildsm committed:
Fix ModelDiscrepancy link in README.md

wildsm committed:
Fix link to ModelDiscrepancy release in examples README

wildsm committed:
Update README for BANDsoftware_uses section

wildsm commented:
Thanks @jared321 for the thorough initial review and @ozgesurer for some changes.

I have tried to address each of @jared321 's points above. In some cases, others have been tagged and should try to chime in soon.

Other reviewers should reviewing the most up-to-date version of the branch.
Thanks all

wildsm committed:
Reword contribution guidelines in README

Clarifying that CONTRIBUTING is the policy.

DanielRPhillips committed:
Update BANDsoftware_uses README to align better with main README

DanielRPhillips committed:
Aligned main README with README in BANDsoftware_uses

DanielRPhillips committed:
Minor tweak to BANDsoftware_uses README to align with main README

wildsm committed:
Update package list formatting in README

Removed links for package descriptions since this is covered in root README. Addresses #188

wildsm committed:
Fix duplicate example entries in README.md

Removed links for examples in README, these are covered in root README. Addresses #188

wildsm committed:
Fix formatting issues in README.md package list

wildsm committed:
Update README to clarify Bfrescox usage

Removing link

DanielRPhillips committed:
Change in capitalization and spacing

DanielRPhillips committed:
Update link to BANDsoftware_uses Bfrescox SDK document

DanielRPhillips committed:
Added BMEX version number

Commit:
Added rose via submodule. v1.1.7 already reviewed, see https://github.com/bandframework/bandframework/pull/167

DanielRPhillips committed:
Update CHANGELOG.rst to include book proposal

wildsm committed:
Fix wording in CHANGELOG about Jupyter notebooks

Fixed spelling caught by actions

DanielRPhillips committed:
Small updates to main README to emphasize bandframework/BAND Framework distinction

DanielRPhillips committed:
Change capitalization to BAND Framework in software directory README

DanielRPhillips committed:
Update README.md for BANDsoftware_uses to give it a header

DanielRPhillips committed:
Link to main README in Software uses README

DanielRPhillips committed:
Update README.md for examples directory to link to main README

DanielRPhillips committed:
Added link to main README and discussion of submodules to Software folder README

DanielRPhillips committed:
Added discussion of submodule updates to main README

DanielRPhillips committed:
Update .wordlist.txt to include submoduled

DanielRPhillips committed:
Update README.md to fix spelling issues

jared321 committed:
Moses prefers LCGP to lcgp

jared321 committed:
Moses prefers LCGP to lcgp

jared321 committed:
Moses prefers LCGP to lcgp

jared321 committed:
Match example name to submodule name

jared321 committed:
Match example name to submodule name

asemposki commented:
I will be adding my full review of this PR tomorrow, just a heads up.

asemposki committed:
updating SAMBA url

asemposki committed:
update SAMBA in changelog

asemposki committed:
Update SaMBA version from v1.2.0 to v1.2.1

DanielRPhillips reviewed (commented):
I have checked the notebooks contained in BANDsoftware_uses/Bfrescox. Three out of four notebooks in BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp_nbs do **not** run at present.

Minimally, the path bandframework/software/Bfrescox/makefiles/googlecolab/makefile Frescox/source/ which occurs in three files needs to be changed to bandframework/BANDsoftware_uses/Bfrescox/makefiles/googlecolab/makefile Frescox/source/

The notebook https://github.com/bandframework/bandframework/blob/v05release/BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp_nbs/Bfrescox_GPR.ipynb can be run successfully, since it does not attempt FRESCOX installation, and so does not need to define any paths. 

@ozgesurer : are you willing to fix the paths and retest?

I also attempted to run the notebook https://github.com/bandframework/bandframework/blob/v05release/examples/BRICK/tutorial/tutorial.ipynb. However, it is explicitly stated that this tutorial does not run correctly unless AZURE2 is installed at the command line. The BRICK README further states:  "AZURE2 is currently only available for the Linux-compatible version. Windows and macOS versions only support the GUI and are therefore inaccessible to BRICK." Therefore I cannot run this test on either of my macOS machines. I suggest we add this as a "Known issue" to the bandframework/CHANGELOG.rst.

I will continue my review on Monday.

asemposki reviewed (changes requested):
I have reviewed the changes; here are my comments.

1) I see that there is an issue currently when looking at the `sdk policies` README.md file; the BMEX and brick sdk links are not working on the `v05release` branch. It looks like these links were not updated when the software was moved to the `examples` folder. The other links all work correctly. I can add a commit to fix this now.

2) For the checking of the BRICK notebook, I ran into the wall of not having AZURE2 and registered for it so I will have to wait until I get it to come back, but the sampling otherwise was set up and was able to correctly initialize before running into that. I however will probably run into the same problem that @DanielRPhillips has mentioned since I am testing on macOS. 

3) For the Bfrescox notebooks, I had the same problems as @DanielRPhillips has also mentioned in his review with the three notebooks requiring Frescox to be installed---it looks like it's pointing to the wrong path. I also was not able to run the GPR notebook all the way through with the latest numpy version of `2.4.0` because the `np.float` option that `ptemcee` requires was deprecated in version `1.20` and has now been removed. I may have missed it but if it has not been noted in the Bfrescox documentation it should be known that numpy must be an older version before this feature was removed, since this is a problem with `ptemcee` and not Bfrescox. I tried to run this again on an older env and it works completely; my Python version there is `3.9.16` and my numpy version there is `1.23.5`, for reference.

asemposki committed:
fixing the BMEX and BRICK links

asemposki commented:
I have fixed issue 1 in my review above so the links now work.

wildsm committed:
Update wordlist

Cleaned up various entries in the wordlist.

wildsm committed:
Updated the wordlist

Readded Özge

wildsm committed:
Change release date for version 0.5.0

Updated lower bound for release date for version 0.5.0.

ozgesurer committed:
Adding a new notebook for BAND Camp 2021

ozgesurer committed:
Merge pull request #193 from bandframework/band21_nb_fix

Adding a new notebook for BAND Camp 2021

wildsm committed:
Add 'ptemcee' to the wordlist

ozgesurer committed:
Update BAND Camp 21 nbs by cloning BAND v0.2

ozgesurer committed:
Merge pull request #194 from bandframework/band21_nb_fix2

Update BAND Camp 21 nbs by cloning BAND v0.2

ozgesurer commented:
- Because the ptemcee sampler is not compatible with the current numpy, I created a new notebook, BANDCamp_nbs/Bfrescox_GPR_v05.ipynb, and illustrated the workflow using emcee. A note has been added to BANDCamp21.rst. See commit `6c5e25e`

- As mentioned previously, the BAND Camp 2021 illustrations should reflect the state of 2021, not 2025. Since we didn’t have a tagged BAND Framework version in 2021 (the closest is v0.2), in the three notebooks that use Fresco I now git clone --branch v0.2.0 https://github.com/bandframework/bandframework.git instead of the develop branch. This resolved the issues. You can now feel free to restructure or update for newer framework versions as needed. See commit `f1bdde6`

DanielRPhillips committed:
Update README.rst

DanielRPhillips committed:
Update CHANGELOG.rst to describe BRICK-AZURE2 compatibility requirements

DanielRPhillips reviewed (commented):
Thanks for the fixes @asemposki and @ozgesurer. @asemposki, can you please check that the revised Bfrescox notebooks now run in Google colab?

I have: 
- updated the README file in the directory where the notebooks sit to clarify the distinction between [Bfrescox_GPR.ipynb](https://github.com/bandframework/bandframework/blob/v05release/BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp_nbs/Bfrescox_GPR.ipynb) and [Bfrescox_GPR_v05.ipynb](https://github.com/bandframework/bandframework/blob/v05release/BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp_nbs/Bfrescox_GPR_v05.ipynb).
- added a known issue to CHANGELOG.rst that states our BRICK MacOS & Windows compatibility issues.

DanielRPhillips committed:
Minor update to examples/README.md

DanielRPhillips reviewed (commented):
@wildsm I have verified several, but not all, of the following items on your list:

- [X]  Release notes for this version are complete and represent changes through the CHANGELOG.rst </CHANGELOG.rst>_ file.

- [X] Ensure that links and references have been updated (e.g., no occurrences of privateband). Consider checking that the url checking action is reviewing a sufficiently rich set of files where spelling should be checked.

**I checked and found no occurences of privateband. I checked several links and, after our efforts of the past few days, found no failures. I also checked that all links in [CONTRIBUTING.rst](https://github.com/bandframework/bandframework/blob/v05release/CONTRIBUTING.rst) and [CHANGELOG.rst](https://github.com/bandframework/bandframework/blob/v05release/CHANGELOG.rst).**

- [X] Version number is update-to-date wherever it appears and +dev suffix is removed
(currently: 4 in README.md </README.md>, 1 in AUTHORS </AUTHORS>, 1 in bandsdk.md </resources/sdkpolicies/bandsdk.md>, 1 in examples </examples/README.md>, 1 in software </software/README.md>, 1 in BANDsoftware_uses </BANDsoftware_uses/README.md>, and possibly in CHANGELOG.rst </CHANGELOG.rst>_).

**Note for future reference that I found one more occurrence of version number in README.md**

- [X] Check README.md </README.md>_ Citing bandframework for correctness (e.g., ensure that author list matches AUTHORS </AUTHORS>_, year and version are correct).

- [X] Check bandsdk.md </resources/sdkpolicies/bandsdk.md>_ Citing bandframework for correctness (authors, year, version). Ensure that SDK version listed is the same as in the provided template </resources/sdkpolicies/template.md>_.

- [X] Check sdkpolicies README </resources/sdkpolicies/README.md>_ for correctness (e.g., ensure that each software in the release has a working link for its SDK compliance, also note any differences in the version of the SDK that each package lists versus the version at bandsdk.md </resources/sdkpolicies/bandsdk.md>_).

**All links work and all packages are working with v0.2 (the current version) of the SDK**

- [ ] Please state whether you have verified successful execution of the notebooks linked in BRICK </examples/BRICK/tutorial>_ and Bfrescox use </BANDsoftware_uses/BFRESCOX>_ . Successful completion of this PR should include verification across the union of reviewers.

**I have not executed BRICK notebooks, but have added issues with doing so as a "Known issue" in [CHANGELOG.rst](https://github.com/bandframework/bandframework/blob/v05release/CHANGELOG.rst)**



- [x] Tests are run with source to be released (this may iterate):

    Online CI (GitHub Actions) tests must pass.

    Documentation must build and display correctly wherever hosted.

ozgesurer committed:
Update README.md

Adding/removing some commas to the README

ozgesurer committed:
Update README.md

Update BFRESCOX --> Bfrescox in the README

ozgesurer committed:
Update README.md

Update PUQ readme for bfrescox path

beykyle reviewed (approved):
I checked the links and didn't find any issues. 

One question: The notebooks linked in https://github.com/bandframework/bandframework/blob/v05release/BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp21.rst now open directly to the git blob for the ipynb files, rather than opening in google collab, is that intentional? @ozgesurer

jared321 committed:
Link points to release branch instead of develop

This should maintain the link valid even after `develop` is eventually updated.  The BRICK notebook link is already set to the release branch.

DanielRPhillips reviewed (approved):
I have completed my review. The only outstanding issue pertains to the notebooks.

jared321 commented:
Review performed at (clean) commit `2c4a4bd5f4`

- [x] Release notes for this version are complete and represent changes through the `CHANGELOG.rst` file.
- [x] Ensure that links and references have been updated (e.g., no occurrences of ``privateband``). Consider checking that the url checking action is reviewing a sufficiently rich set of files where spelling should be checked.
- [x] Version number is update-to-date wherever it appears and ``+dev`` suffix is removed
  **(currently: 3 in `README.md </README.md>`_, 1 in `AUTHORS </AUTHORS>`_, 1 in `bandsdk.md </resources/sdkpolicies/bandsdk.md>`_, 1 in `examples </examples/README.md>`_, 1 in `software </software/README.md>`_, 1 in `BANDsoftware_uses </BANDsoftware_uses/README.md>`_, and possibly in `CHANGELOG.rst </CHANGELOG.rst>`_)**.
- [x] Check `README.md` *Citing bandframework* for correctness (e.g., ensure that author list matches `AUTHORS`, **year and version are correct**).
- [x] Check `bandsdk.md` *Citing bandframework* for correctness **(authors, year, version)**. Ensure that SDK version listed is the same as in the provided `template`.
- [x] Check `sdkpolicies README` for correctness (e.g., ensure that each software in the release has a working link for its SDK compliance, also note any differences in the version of the SDK that each package lists versus the version at `bandsdk.md`).
* Please state whether you have verified successful execution of the notebooks linked in
  - [ ] `BRICK` - I don't have AZURE2 on a Linux station
  - [x] `Bfrescox + surmise` - Built Frescox locally, setup dedicated venv with manual installation of external dependencies , and ran `Bfrescox_GPR_v05.ipynb`.  Plots are similar to those in repo.  Also manually updated `Bfrescox_surmise.ipynb` to use my local installations (including Frescox binary) and confirm that it ran through with graphics similar to those in that notebook.
* Tests are run with source to be released (this may iterate):
  - [x] Online CI (GitHub Actions) tests must pass.
  - [ ] Documentation must build and display correctly wherever hosted.
  
Additionally

* Confirmed that all versions provided in README for this release are identical to versions listed in
   - [x] `software/README.md`
   - [x] `examples/README.md` 
   - [x] `BANDsoftware_uses/README.md` (N/A)

wildsm committed:
Linking to rose bandsdk

For completeness

wildsm committed:
Add 'rosebandsdk' to wordlist

wildsm committed:
Update URL checker blacklist in CI workflow

jared321 committed:
Modernize Bfrescox tutorial notes for release

The tutorial notebooks can no longer hosted on the original Colab-compatible site.  Make this clear in the notes and provide some hints for how to update the notebooks for local execution.

jared321 commented:
I updated the Bfrescox+surmise tutorial README to address potential notebook issues/questions raised by both Daniel and Kyle.

3b330e4

ozgesurer commented:
> I checked the links and didn't find any issues.
> 
> One question: The notebooks linked in https://github.com/bandframework/bandframework/blob/v05release/BANDsoftware_uses/Bfrescox/Tutorial_I/BANDCamp21.rst now open directly to the git blob for the ipynb files, rather than opening in google collab, is that intentional? @ozgesurer

We decided to keep the files themselves so that we do not lose access to the links in case of any reason. Please also see the commit `3b330e4` for the updated explanation.

jared321 reviewed (approved):
Please see my review checklist in the PR notes.

wildsm commented:
Many thanks to all of the reviewers!

We have come to consensus that @asemposki 's notes were addressed and are proceeding with next steps since she is likely  logged off for the year.

I will continue following the flow at https://github.com/bandframework/bandframework/issues/178
